src/EditorFeatures/Core/Shared/Extensions/ITextBufferEditExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Extensions
+{
+    internal static class ITextBufferEditExtensions
+    {
+        private static Exception s_lastException = null;
+
+        /// <summary>
+        /// Logs exceptions thrown during <see cref="ITextBufferEdit.Apply"/> as we look for issues.
+        /// </summary>
+        /// <param name="edit"></param>
+        /// <returns></returns>
+        public static ITextSnapshot ApplyAndLogExceptions(this ITextBufferEdit edit)
+        {
+            try
+            {
+                return edit.Apply();
+            }
+            catch (Exception e) when (ErrorReporting.FatalError.ReportWithoutCrash(e))
```

Comment:
Today this exception is caught by the editor and reported to the user.  Do we not report anything if the editor shows a dialog?

---
In reply to: [120682599](https://github.com/dotnet/roslyn/pull/20063#discussion_r120682599) [](ancestors = 120682599)